keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -128,16 +128,30 @@ class ImageService(
     appVersion: AppVersion,
     account: String,
     region: String
-  ): NamedImage? =
-    namedImageCache
-      .get(NamedImageCacheKey(appVersion, account, region))
-      .await()
+  ): NamedImage? {
+    val key = NamedImageCacheKey(appVersion, account, region)
+    return namedImageCache.get(key).await()
       // prime the cache if the image is also in other regions
       ?.also { image ->
         (image.amis.keys - region).forEach { otherRegion ->
-          namedImageCache.put(NamedImageCacheKey(appVersion, account, otherRegion), completedFuture(image))
+          namedImageCache.putIfMissingOrOlderThan(key.copy(region = otherRegion), image)
         }
       }
+  }
+
+  private fun AsyncCache<NamedImageCacheKey, NamedImage>.putIfMissingOrOlderThan(
+    key: NamedImageCacheKey,
+    image: NamedImage
+  ) {
+    synchronous().apply {
+      val updateCache = getIfPresent(key)
+        ?.let { it.creationDate < image.creationDate }
+        ?: true
+      if (updateCache) {
+        put(key, image)
+      }
+    }
+  }
```

Comment:
This is the implementation of that logic. It operates on a `synchronous` view of the cache as it's using `getIfPresent` to avoid calling CloudDriver so there's no need to handle the result asynchronously.